src/routes/boot.js

```
@@ -130,41 +130,46 @@ export const bootSharedLogic = async (ctx, shouldRefreshToken) => {
     const userRequests = isKratos
       ? [getUserFromAPI(ctx), [], []]
       : [userModel.getById(userId), provider.getByUserId(userId), role.getByUserId(userId)];
-    const [user, userProvider, roles] = await Promise.all(userRequests);
-
-    if (!user) {
-      setTrackingId(ctx, null);
-      throw new ForbiddenError();
-    }
-
-    const accessToken = shouldRefreshToken
-      ? await setAuthCookie(ctx, user.id, roles)
-      : undefined;
+    try {
+      const [user, userProvider, roles] = await Promise.all(userRequests);
+      if (!user) {
+        setTrackingId(ctx, null);
+        throw new ForbiddenError();
+      }
 
-    const base = await bootBaseResponse(
-      ctx,
-      visitId,
-      visitPromise,
-      now,
-      referral,
-      user,
-    );
-    returnObject = {
-      ...base,
-      user: {
-        ...base.user,
-        ...user,
-        providers: [userProvider.provider],
-        roles,
-        permalink: `${config.webappOrigin}/${user.username || user.id}`,
-      },
-      accessToken,
-    };
-    if (!user.infoConfirmed) {
+      const accessToken = shouldRefreshToken
+        ? await setAuthCookie(ctx, user.id, roles)
+        : undefined;
+
+      const base = await bootBaseResponse(
+        ctx,
+        visitId,
+        visitPromise,
+        now,
+        referral,
+        user,
+      );
       returnObject = {
-        ...returnObject,
-        registrationLink: `${config.webappOrigin}/register`,
+        ...base,
+        user: {
+          ...base.user,
+          ...user,
+          providers: [userProvider.provider],
+          roles,
+          permalink: `${config.webappOrigin}/${user.username || user.id}`,
+        },
+        accessToken,
       };
+      if (!user.infoConfirmed) {
+        returnObject = {
+          ...returnObject,
+          registrationLink: `${config.webappOrigin}/register`,
+        };
+      }
+    } catch (error) {
```

I think the catch here is too wide. Can we scope it down to a specific set of errors?
I see that we also throw forbidden in case the user can't be found. Maybe it's better to just log out the user and continue as usual?

---

We don't really get more info back at this point.
It's a generic graphQL error thrown from the API.

But agree perhaps we can simply log out and return the default user?

---

Yes, the anonymous user works.
But what exactly is the GraphQL error? Maybe we should return a specific error. Because, if the API doesn't have access to the DB at the moment for some reason we don't want to think the user doesn't exist. Wide catch makes us blind to unexpected errors

---

@idoshamun I've added an auto logout for when users are not found, this will return the default base object + a new `shouldLogout` flag, (we can choose to use it in the frontend later on).

I think the only edge case we need to sort is the errors from the API.
We should only logout on specific ones, other could simply throw some kind of error (API down, for instance)

In my opinion person in the catch should only be logged-out if they user is not found or malformed, any other should throw an error. 
(Let me know what you think)

---

I agree. Any other error should be thrown which will yield unexpected error (which is exactly what we want)

---

I'll try and add those specific errors from the API.